lib/sign_in/idme/configuration.rb

```diff
@@ -71,7 +71,19 @@ def jwt_decode_algorithm
       end
 
       def jwt_decode_public_key
-        OpenSSL::PKey::RSA.new(File.read(Settings.idme.oauth_public_key))
+        IdmePublicKey.public_keys ||= fetch_idme_public_keys
```

Comment:
These Configuration classes are actually already set up as a `Singleton` object, so probably the easiest way to do this is actually just to store a variable on this class `@idme_public_keys ||= fetch_idme_public_keys`.